geode-core/src/main/java/org/apache/geode/internal/cache/execute/AbstractExecution.java

```diff
@@ -435,7 +447,7 @@ public boolean isIgnoreDepartedMembers() {
     return ignoreDepartedMembers;
   }
 
-  protected abstract ResultCollector executeFunction(Function fn);
+  protected abstract ResultCollector executeFunction(Function fn, int timeoutMs);
```

Comment:
Changing this method to take the timeout and unit would remove the duplicate conversions from time an unit to milliseconds. Also in that conversion you need to account for values less than 1ms, like 1ns, as well as values greater than `Integer.MAX_VALUE`ms, like `Long.MAX_VALUE`ms. I would suggest a little timeout helper class to do this since we will need it later.

Reply:
I added the TimeUnit parameter to the executeFunction method but still there are several places where the conversion must be done.
I have also added the helper class suggested.
Thanks!